### PR TITLE
Handle push_events_branch_filter in gitlab hooks

### PR DESCRIPTION
**Pull request by Turgon37**
##### SUMMARY
Gitlab version 11.3 introduce the push_events_branch_filter key into hook api endpoint. The gitlab_hook module should support it

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
gitlab_hook

### CODE REVIEW

File: lib/ansible/modules/source_control/gitlab_hook.py [hunk 1]
```diff
@@ -309,6 +316,7 @@ def main():
         project=dict(type='str', required=True),
         hook_url=dict(type='str', required=True),
         push_events=dict(type='bool', default=True),
+        push_events_branch_filter=dict(type='str'),
```
Comment (Lunik) [inline lib/ansible/modules/source_control/gitlab_hook.py]: You should maybe set the default value to `""`

Comment (Turgon37) [reply lib/ansible/modules/source_control/gitlab_hook.py]: Thanks for your remark, without the default value the module was not doing field's cleaning if the parameter was not present. Now, with it, it does

**Comment by Shaps** [comment]
@Turgon37 Looks good to me, can you close/reopen so CI can run again?

**Comment by Lunik** [comment]
@Turgon37 can you please rebase your PR

**Review by Lunik — CHANGES_REQUESTED** [review]
Update the `version_added` to `2.9`

**Comment by vvanouytsel** [comment]
Should we re-run shippable, seems to be stuck with a pending check?

**Comment by Turgon37** [comment]
Hello, I've just to rebase the pr :)

**Comment by Lunik** [comment]
shipit

**Comment by vvanouytsel** [comment]
Could we also backport this to 2.9?

**Comment by Lunik** [comment]
> Could we also backport this to 2.9?

No, we don't backport new features

**Comment by Shaps** [comment]
shipit

**Pull request merged**